std/algorithm/searching.d

```diff
@@ -1204,6 +1210,75 @@ bool endsWith(alias pred, R)(R doesThisEnd)
     }
 }
 
+/**
+Iterates the passed range and selects the extreme element with $(D less).
+If the extreme element occurs multiple time, the first occurrence will be
+returned.
+
+Params:
+    map = custom accessor for the comparison key
+    selector = custom mapping for the extrema selection
+    r = Range from which the extreme value will be selected
+
+Returns: The extreme value according to $(D less) of the passed-in values.
+*/
+private auto extremum(alias map = "a", alias selector = "a < b", Range)(Range r)
+    if (isInputRange!Range && !isInfinite!Range)
+in
+{
+    assert(!r.empty, "r is an empty range");
+}
+body
+{
+    alias mapFun = unaryFun!map;
+    alias selectorFun = binaryFun!selector;
+
+    // init extremum
+    Unqual!(ElementType!Range) extremeElement = r.front;
```

Comment:
Nitpick - how about:
```
alias Element = ElementType!Range;
alias MapType = Unqual!(typeof(mapFun(Element.init)));

Unqual!Element extremeElement = r.front;
MapType extremeElementMapped = mapFun(extremeElement);
```
(Evaluate `r[0]` once and make the code a bit more DRY).